pkg/controller/cronjob/utils.go

```diff
@@ -147,6 +150,65 @@ func getRecentUnmetScheduleTimes(cj batchv1beta1.CronJob, now time.Time) ([]time
 	return starts, nil
 }
 
+// getUnmetScheduleTimes gets the slice of all the missed times from the time a job
+// last was scheduled to up `now`.
+//
+// If there are too many (>100) unstarted times, it will raise a warning and but still return
+// the list of missed times.
+func getUnmetScheduleTimes(cj batchv1beta1.CronJob, now time.Time, schedule cron.Schedule, recorder record.EventRecorder) []time.Time {
+	starts := []time.Time{}
+
+	var earliestTime time.Time
+	if cj.Status.LastScheduleTime != nil {
+		earliestTime = cj.Status.LastScheduleTime.Time
+	} else {
+		// If none found, then this is either a recently created cronJob,
+		// or the active/completed info was somehow lost (contract for status
+		// in kubernetes says it may need to be recreated), or that we have
+		// started a job, but have not noticed it yet (distributed systems can
+		// have arbitrary delays).  In any case, use the creation time of the
+		// CronJob as last known start time.
+		earliestTime = cj.ObjectMeta.CreationTimestamp.Time
+	}
+	if cj.Spec.StartingDeadlineSeconds != nil {
+		// Controller is not going to schedule anything below this point
+		schedulingDeadline := now.Add(-time.Second * time.Duration(*cj.Spec.StartingDeadlineSeconds))
+
+		if schedulingDeadline.After(earliestTime) {
+			earliestTime = schedulingDeadline
+		}
+	}
+	if earliestTime.After(now) {
+		return []time.Time{}
+	}
+
+	for t := schedule.Next(earliestTime); !t.After(now); t = schedule.Next(t) {
+		starts = append(starts, t)
+	}
+	if len(starts) > 100 {
```

Comment:
Can we please get rid of this 'feature'?

https://github.com/kubernetes/kubernetes/pull/81557 is still open to fix this in v1 controller, but I think it would be best to fix this in the v2 controller right from the start.

Reply:
Yeah I was talking about it with @alaypatel07 it should not matter for the v2 controller, though.